src/common/filesystembase.cpp

```diff
@@ -621,7 +621,9 @@ bool FileSystem::isChildPathOf(QStringView child, QStringView parent)
     }
 
     // Slow path (`QDir::cleanPath` does lots of string operations):
-    return QString::compare(QDir::cleanPath(parent.toString()), QDir::cleanPath(child.toString()), sensitivity) == 0;
+    // cleanPath removes trailing slashes, add one to parent to be sure we handle a child path
+    // root/foo/bar is not  a child of /root/fo, therefor the trailing slash is important
+    return QDir::cleanPath(child.toString()).startsWith(QDir::cleanPath(parent.toString()) + QLatin1Char('/'), sensitivity);
```

Comment:
Looks like adding a slash to both of them works: for `/a/b/c/` for child and path as an input, I get the correct result from this snippet:

```cpp
    const QString parent("/a/b/c/");
    const QString child("/a/b/c/");
    qDebug() << QDir::cleanPath(child).startsWith(QDir::cleanPath(parent), Qt::CaseSensitivity::CaseSensitive);

Reply:
Needs an exception for files: `!parent.isFile() && [...]`. Files can never have children.